packages/loot-core/src/client/queries/queriesSlice.ts

```diff
@@ -0,0 +1,867 @@
+
+type UpdateCategoryPayload = {
+  category: CategoryEntity;
+};
+
+export const updateCategory = createAppAsyncThunk(
+  `${sliceName}/updateCategory`,
+  async ({ category }: UpdateCategoryPayload, { dispatch }) => {
+    await send('category-update', category);
+    await dispatch(getCategories());
+  },
+);
+
+type DeleteCategoryPayload = {
+  id: CategoryEntity['id'];
+  transferId?: CategoryEntity['id'];
+};
+
+export const deleteCategory = createAppAsyncThunk(
+  `${sliceName}/deleteCategory`,
+  async ({ id, transferId }: DeleteCategoryPayload, { dispatch }) => {
+    const { error } = await send('category-delete', { id, transferId });
+
+    if (error) {
+      switch (error) {
+        case 'category-type':
+          dispatch(
+            addNotification({
+              type: 'error',
+              message: t(
+                'A category must be transferred to another of the same type (expense or income)',
+              ),
+            }),
+          );
+          break;
+        default:
+          dispatch(addGenericErrorNotification());
+      }
+
+      throw new Error(error);
+    } else {
+      await dispatch(getCategories());
+      // Also need to refresh payees because they might use one of the
+      // deleted categories as the default category
+      await dispatch(getPayees());
+    }
+  },
+);
+
+type MoveCategoryPayload = {
+  id: CategoryEntity['id'];
+  groupId: CategoryGroupEntity['id'];
+  targetId: CategoryEntity['id'];
+};
+
+export const moveCategory = createAppAsyncThunk(
+  `${sliceName}/moveCategory`,
+  async ({ id, groupId, targetId }: MoveCategoryPayload, { dispatch }) => {
+    await send('category-move', { id, groupId, targetId });
+    await dispatch(getCategories());
+  },
+);
+
+type MoveCategoryGroupPayload = {
+  id: CategoryGroupEntity['id'];
+  targetId: CategoryGroupEntity['id'];
+};
+
+export const moveCategoryGroup = createAppAsyncThunk(
+  `${sliceName}/moveCategoryGroup`,
+  async ({ id, targetId }: MoveCategoryGroupPayload, { dispatch }) => {
+    await send('category-group-move', { id, targetId });
+    await dispatch(getCategories());
+  },
+);
+
+export const getCategories = createAppAsyncThunk(
+  `${sliceName}/getCategories`,
+  async () => {
+    const categories: CategoryViews = await send('get-categories');
+    return categories;
+  },
+);
+
+// Payee actions
+
+type CreatePayeePayload = {
+  name: PayeeEntity['name'];
+};
+
+export const createPayee = createAppAsyncThunk(
+  `${sliceName}/createPayee`,
+  async ({ name }: CreatePayeePayload, { dispatch }) => {
+    const id: PayeeEntity['id'] = await send('payee-create', {
+      name: name.trim(),
+    });
+    dispatch(getPayees());
+    return id;
+  },
+);
+
+export const initiallyLoadPayees = createAppAsyncThunk(
+  `${sliceName}/initiallyLoadPayees`,
+  async (_, { dispatch, getState }) => {
+    const queriesState = getState().queries;
+    if (queriesState.payees.length === 0) {
+      return dispatch(getPayees());
```

Comment:
This may need `.unwrap` as well

Reply:
Checked the callers of this thunk and confirmed that they don't depend on the result of it so I just remove the return statement